docs/developers_notes/06-glm.md

```diff
@@ -0,0 +1,91 @@
+
+Additionally, the `GLM` class inherits the attributes of `BaseRegressor`, see the [relative note](03-base_regressor.md) for more information.
+
+### Public Methods
+
+- **`predict`**: Validates input and computes the mean rates of the `GLM` by invoking the inverse-link function of the `observation_models` attribute.
+- **`score`**: Validates input and assesses the Poisson GLM using either log-likelihood or pseudo-$R^2$. This method uses the `observation_models` to determine log-likelihood or pseudo-$R^2$.
+- **`fit`**: Validates input and aligns the Poisson GLM with spike train data. It leverages the `observation_models` and `regularizer` to define the model's loss function and instantiate the regularizer.
+- **`simulate`**: Simulates spike trains using the GLM as a feedforward network, invoking the `observation_models.sample_generator` method for emission probability.
+- **`initialize_params`**: Initialize model parameters, setting to zero the coefficients, and setting the intercept by matching the firing rate.
+- **`initialize_state`**: Initialize the state of the solver.
+- **`update`**: Run a step of optimization and update the parameter and solver step.
+
+### Private Methods
+
+Here we list the private method related to the model compuations:
+
+- **`_predict`**: Forecasts rates based on current model parameters and the inverse-link function of the `observation_models`.
+- **`_predict_and_compute_loss`**: Predicts the rate and calculates the mean Poisson negative log-likelihood, excluding normalization constants.
+
+A number of `GLM` specific private methods are used for checking parameters and inputs, while the methods related for checking the solver-regularizer configurations/instantiation are inherited from `BaseRergessor`.
+
+
+## The Concrete Class `PopulationGLM`
+
+The `PopulationGLM` class is an extension of the `GLM`, designed to fit multiple neurons jointly. This involves vectorized fitting processes that efficiently handle multiple neurons simultaneously, leveraging the inherent parallelism.
+
+### `PopulationGLM` Specific Attributes
+
+- **`feature_mask`**: A mask that determines which features are used as predictors for each neuron. It can be a matrix of shape `(num_features, num_neurons)` or a `FeaturePytree` of binary values, where 1 indicates that a feature is used for a particular neuron and 0 indicates it is not.
+
+### Overridden Methods
+
+- **`fit`**: Overridden to handle fitting of the model to a neural population. This method validates input including the mask and fits the model parameters (coefficients and intercepts) to the data.
+- **`_predict`**: Computes the predicted firing rates using the model parameters and the feature mask.
+
+
+
+## Contributor Guidelines
+
+### Implementing GLM Subclasses
+
+When crafting a functional (i.e., concrete) GLM class:
+
+- **Must** inherit from `BaseRegressor` or one of its derivatives.
+- **Must** realize all abstract methods.
```

Comment:
this page says nothing about abstract methods (because GLM and PopulationGLM don't have any). I'd say explicitly: "if you inherit `GLM` or any of the other concrete classes, there won't be any abstract methods. See [BaseRegressor page](link) for details on its abstract methods if you want to inherit it directly`.